api/scripts/weekly-digest.ts

```diff
@@ -0,0 +1,125 @@
+import { SMTPClient } from "emailjs";
+import { createEmailWithTemplate } from "../emails/htmlEmailUtils.js";
+import config from "@config";
+import type { MonitoringParams } from "../api/V1/monitoringPage.js";
+import { calculateMonitoringPageCriteria } from "../api/V1/monitoringPage.js";
+import { generateVisits } from "../api/V1/monitoringVisit.js";
+import modelsInit from "@models/index.js";
+import { RecordingType } from "@typedefs/api/consts.js";
+
+(async () => {
+  try {
+    const models = await modelsInit();
+    const users = await models.User.findAll({where: {
+      'settings.emailNotifications.weeklyDigest': true
+    }});
+    
+    const templateFilename = "weekly-digest.html";
+    const client = new SMTPClient(config.smtpDetails);
+    // const client = new SMTPClient({
+    //   user: "",
+    //   password: "",
+    //   host: "smtp.gmail.com",
+    //   ssl: true,
+    // });
+
+    users.forEach(async user => {
```

Comment:
Given that weekly and daily digests share the exact same logic, and only differ by their timespan, maybe pull the shared logic out into a helper function?